File: RB101_RB109_small_problems/easy_4/convert_a_number_to_a_string.rb
DIGITS = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']

def integer_to_string(num)
  strings = num.digits.reverse.map {|num| DIGITS[num]}
  strings.join("")
end

p integer_to_string(4321) == '4321'
p integer_to_string(0) == '0'
p integer_to_string(5000) == '5000'

#Further Exploration
=begin
Mutating String methods not ending in ! include:
  []=
  concat
  replace
  insert
  
Mutating Array methods not ending in ! include:
  []=
  append
  push
  <<
  unshift
  pop
  prepend
  shift
  insert
  delete
  delete_at
  delete_if
  keep_if
  
Mutating Hash methods not ending in ! include:
  []=
  clear
  delete
  delete_if
  keep_if
  update
  rehash
  shift
=end

File: JS210_fundamentals/basics/convert_a_number_to_a_string.js
function integerToString(num) {
  const BASE = 10;
  const DIGITS_TO_STRING = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
  let stringComponents = [];

  do {
    let digit = num % BASE;
    stringComponents.unshift(DIGITS_TO_STRING[digit]);
    num = Math.floor(num / BASE);
  } while (num > 0);

  return stringComponents.join('');
}

console.log(integerToString(4321));      // "4321"
console.log(integerToString(0));         // "0"
console.log(integerToString(5000));      // "5000"